src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -28476,6 +28476,263 @@ void M2()
+            comp.VerifyDiagnostics(
+                // (14,12): warning CS8604: Possible null reference argument for parameter 's' in 'void C.M1(string s)'.
+                //         M1((C?)null); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "(C?)null").WithArguments("s", "void C.M1(string s)").WithLocation(14, 12));
+        }
+
+        [Fact]
+        [WorkItem(39802, "https://github.com/dotnet/roslyn/issues/39802")]
+        public void NotNullIfNotNull_ExplicitUserDefinedOperator()
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    [return: NotNullIfNotNull(""c"")]
+    public static explicit operator string?(C? c) => c?.ToString();
+
+    void M1(string s) { }
+
+    void M2()
+    {
+        M1((string)new C());
+        M1((string?)(C?)null); // 1
+    }
+}
+";
+            var comp = CreateNullableCompilation(new[] { NotNullIfNotNullAttributeDefinition, source });
+            comp.VerifyDiagnostics(
+                // (13,12): warning CS8604: Possible null reference argument for parameter 's' in 'void C.M1(string s)'.
+                //         M1((string?)(C?)null); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "(string?)(C?)null").WithArguments("s", "void C.M1(string s)").WithLocation(13, 12));
+        }
+
+        [Fact]
+        [WorkItem(39802, "https://github.com/dotnet/roslyn/issues/39802")]
+        public void UserDefinedConversion_UnconditionalParamAttributes()
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+
+public struct A { }
+public struct B
+{
+    public static implicit operator A([DisallowNull] B? b)
+    {
+        b.Value.ToString();
+        return new A();
+    }
+
+    void M1(A a) { }
+
+    void M2()
+    {
+        M1(new B());
+        M1((B?)null); // 1
+    }
+}
+
+public class C
+{
+    public static implicit operator string?([AllowNull] C c) => c?.ToString();
+
+    void M1(string? s) { }
+    void M2()
+    {
+        M1(new C());
+        M1((C?)null);
+    }
+}
+";
+            var comp = CreateNullableCompilation(new[] { AllowNullAttributeDefinition, DisallowNullAttributeDefinition, source });
+            comp.VerifyDiagnostics(
+                // (18,12): warning CS8607: A possible null value may not be used for a type marked with [NotNull] or [DisallowNull]
+                //         M1((B?)null); // 1
+                Diagnostic(ErrorCode.WRN_DisallowNullAttributeForbidsMaybeNullAssignment, "(B?)null").WithLocation(18, 12));
+        }
+
+        [Fact]
+        [WorkItem(39802, "https://github.com/dotnet/roslyn/issues/39802")]
+        public void UserDefinedConversion_UnconditionalParamAttributes_Lifted()
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+
+public struct A { }
+
+public struct B
+{
+    public static implicit operator A([DisallowNull] B b)
```

Comment:
>[DisallowNull] [](start = 38, length = 14)

What does `[DisallowNull]` mean for a user-defined conversion operand that is a value type?

Reply:
Curious what it means for both the operator and the lifted version of the operator.

Reply:
Here I was trying to verify that the attribute doesn't affect anything, on a non-nullable operator lifted to nullable, and on a nullable operator lifted to return nullable. The only situation where we believe the operator will be called is if the argument is non-nullable, in which case there is nothing for the DisallowNullAttribute to verify.

If these tests seem too inconsequential I can delete them.

Reply:
Think the test is fine and important. The code is indeed non-sense but it's legal for users to write this and we should verify that we behave as expected. I would suggest a comment noting that the intent here is to verify we correctly do nothing special.